### PR TITLE
sometimes stop doesnt stop trace

### PR DESCRIPTION
**Pull request by anporumb**

### CODE REVIEW

File: pipeline_templates/scripts/stop_trace.ps1 [hunk 1]
```diff
@@ -21,8 +21,32 @@ try
...
+
+    # Setup a timer
+    $timer = New-Object System.Timers.Timer
+    $timer.Interval = $timeout
+    $timer.AutoReset = $false
+
+    # Define the action to take when the timer elapses
+    $timerEvent = Register-ObjectEvent -InputObject $timer -EventName Elapsed -Action {
+        Write-Host "Timeout reached. Restarting the machine! Expect the build to (eventually) fail... "
+        Restart-Computer -Force
```
Comment (mattdurak) [inline pipeline_templates/scripts/stop_trace.ps1:38]: We have sysinternals on the gate machines, I wonder if we should bugcheck here with NotMyFault and get a dump to send to the Windows team if the ETW logging is stuck? [https://learn.microsoft.com/en-us/sysinternals/downloads/notmyfault](https://learn.microsoft.com/en-us/sysinternals/downloads/notmyfault)

Just an idea...

**Review by mattdurak — APPROVED** [review]
:shipit:

File: pipeline_templates/scripts/stop_trace.ps1 (hunk 1 above)
Comment (anporumb) [reply pipeline_templates/scripts/stop_trace.ps1:47]: We've talked and decided to procdump + kernel crash.

... something will happen :)

**Review by mattdurak — APPROVED** [review]
:shipit:

**Pull request merged**